pkg/services/secrets/migrator/provisioning.go

```diff
@@ -0,0 +1,187 @@
+package migrator
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/services/encryption"
+	"github.com/grafana/grafana/pkg/services/secrets"
+	"github.com/grafana/grafana/pkg/services/secrets/manager"
+)
+
+var errNoEncryptedValue = errors.New("no encrypted value was found")
+
+var _ SecretsRotator = provisioningSecrets{}
+
+type provisioningSecrets struct{}
+
+func (p provisioningSecrets) ReEncrypt(ctx context.Context, svc *manager.SecretsService, sqlStore db.DB) bool {
+	return p.reEncrypt(ctx, svc, func(ctx context.Context, payload []byte) ([]byte, error) {
+		return svc.Encrypt(ctx, payload, secrets.WithoutScope())
+	}, sqlStore, "rotation")
+}
+
+func (p provisioningSecrets) Rollback(ctx context.Context, svc *manager.SecretsService, internal encryption.Internal, sqlStore db.DB, secretKey string) bool {
+	return p.reEncrypt(ctx, svc, func(ctx context.Context, payload []byte) ([]byte, error) {
+		return internal.Encrypt(ctx, payload, secretKey)
+	}, sqlStore, "rollback")
+}
+
+func (p provisioningSecrets) reEncrypt(
+	ctx context.Context,
+	svc *manager.SecretsService,
+	encrypt func(context.Context, []byte) ([]byte, error),
+	sqlStore db.DB,
+	action string,
+) bool {
+	var rows []struct {
+		Guid  string
+		Value []byte
+	}
+
+	if err := sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
```

Comment:
I am not sure if I follow the strategy.  What happens if our secrets are only stored in unified storage? 

I was expecting some kind of hook that will list all repository and do a patch to update Encrypted Github and Webhook secrets so that also the webhook will be updated with the new secret thanks to our repository controller and `OnUpdate` hook.